src/graphics/opengl/renderer.cpp

```diff
@@ -270,25 +273,42 @@ Renderer::Renderer(Platform::Window &window, const std::string &shaderDirectory)
 		_renderPasses.emplace_back(RenderPass(item.first.programName, item.first.properties));
 	}
 
-	// Get width and height of the default framebuffer
-	unsigned int width, height;
-	window.getSize(width, height);
+	_deferredBuffer = std::make_unique<Framebuffer>();
+	setRenderPlane(_window.getSize());
+
+	// Check for errors
+    assert(glGetError() == GL_NO_ERROR);
+}
+
+void Renderer::setRenderPlane(unsigned int width, unsigned int height) {
+	setRenderPlane(glm::vec2(width, height));
+}
+
+void Renderer::setRenderPlane(const glm::vec2 size) {
+	Graphics::Renderer::setRenderPlane(size);
 
 	// Initialize deferred shading
 	//
-	_depthTexture = std::make_unique<Texture>(width, height);
-	_normalTexture = std::make_unique<Texture>(width, height);
+	if (_depthTexture) {
+		_depthTexture.reset();
+	}
+	if (_normalTexture) {
+		_normalTexture.reset();
+	}
```

Comment:
Both textures are already overwritten automatically by their recreation below, so ou do not need to reset them.